docs/spring-boot/app-starter.md

```diff
@@ -0,0 +1,145 @@
+[Spring Boot](https://spring.io/projects/spring-boot) application.
+ 
+## Features
+- Configure extension app through `application.yaml`
+- Expose a REST API to perform [Circle of Trust authentication](https://developers.symphony.com/extension/docs/application-authentication).
+
+## Installation
+
+The following listing shows the `pom.xml` file that has to be created when using Maven:
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>bdk-app-spring-boot</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>bdk-app-spring-boot</name>
+    
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.symphony.platformsolutions</groupId>
+                <artifactId>symphony-bdk-bom</artifactId>
+                <version>1.3.2.BETA</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.symphony.platformsolutions</groupId>
+            <artifactId>symphony-bdk-app-spring-boot-starter</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+        </dependency>
+    </dependencies>
+    
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-maven-plugin</artifactId>
+                    <version>2.3.4.RELEASE</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
+```
+The following listing shows the `build.gradle` file that has to be created when using Gradle:
+```groovy
+plugins {
+    id 'java-library'
+    id 'org.springframework.boot' version '2.3.4.RELEASE'
+}
+
+dependencies {
+    implementation platform('com.symphony.platformsolutions:symphony-bdk-bom:1.3.2.BETA')
+    
+    implementation 'com.symphony.platformsolutions:symphony-bdk--spring-boot-starter'
+    implementation 'org.springframework.boot:spring-boot-starter'
+}
+```
+
+## Create a Simple Backend for Extension Application
+As a first step, you have to initialize your environment through the Spring Boot `src/main/resources/application.yaml` file: 
+```yaml
+bdk:
+    host: acme.symphony.com
+    bot:
+      username: bot-username
+      privateKeyPath: /path/to/rsa/privatekey.pem
+      
+    app:
```

Comment:
I'm not sure where we want to keep the appId and the privatekey, because the `CoreProperties` extending the BdkConfig so I guess the appId and privatekey should be under the bdk.

Reply:
I just realize that the `ExtensionAppAuthenticator` bean (which is in `core-starter`) will be created if the `appId` is defined, so if we want to move `appId` under `bdk-app`, then the `SymphonyBdkAppProperties` must be declared in `core-starter` instead of `app-starter`

Reply:
`app-starter` depends on `core-starter`. So `appId` etc. have to be configured at `core-starter` level: 
```yaml
bdk: # configure core
   appId: blabla
bdk-app: # configure circle of trust controller only, among other future things (note that this configuration could be optional)
   auth: 
       enabled: true
```

Reply:
It has to be:
```
bdk: # configure core
   app:
     appId: blabla
     privateKey: /path/to/privateKey
bdk-app: # configure circle of trust controller only, among other future things (note that this configuration could be optional)
   auth: 
       enabled: true
```
because the coreProperties will be used as the config passed to the [AuthenticatorFactory](https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/config/BdkCoreConfig.java#L63-L67)